Lib/test/test_zipfile/_path/test_path.py

```diff
@@ -577,3 +577,15 @@ def test_getinfo_missing(self, alpharep):
         zipfile.Path(alpharep)
         with self.assertRaises(KeyError):
             alpharep.getinfo('does-not-exist')
+
+    def test_root_folder_in_zipfile(self):
+        """
+        gh-112795: Some tools or self constructed codes will add '/' folder to
+        the zip file, this is a strange behavior, but we should support it.
+        """
+        in_memory_file = io.BytesIO()
+        zf = zipfile.ZipFile(in_memory_file, "w")
+        zf.mkdir('/')
+        zf.writestr('./a.txt', 'aaa')
+        tmpdir = pathlib.Path(self.fixtures.enter_context(temp_dir()))
+        zf.extractall(tmpdir)
```

Comment:
This test is in the wrong file. Everything under the `_path` path is meant to test the `zipfile.Path` class, which this test doesn't. I'll move the test to the appropriate module.

It might be worth exploring what effect a root dir would have on the zipfile.Path, however.